app/components/footnote_modal_component.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+class FootnoteModalComponent < ViewComponent::Base
+  renders_one :body_content
+
+  def initialize(title:)
+    @title = title
+  end
+
+  def icon
+    helpers.fa_icon('question-circle')
+  end
+
+  def modal_id
+    'footnote-modal-' + object_id.to_s
```

Comment:
When the model is added to a page, other elements will need to reference its id -- the bootstrap support requires having another element that refers to it. So I think it would be better to have the id specified in the constructor as a required parameter.